### PR TITLE
Reduce the resource consumption of flaky tests

### LINKED ISSUE #3742: 'Expiring Daemon because JVM heap space is exhausted' error in flaky-tests
See: https://github.com/line/armeria/runs/3236075032#step:4:2208

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

Our 'flaky-tests' job sometimes fails with the following error:

    Expiring Daemon because JVM heap space is exhausted

It's somewhat unclear why this happens to only 'flaky-tests' job, but we
could work around this by adjusting the build parameters.

Modifications:

- Update the max JVM heap size of the Gradle daemon.
- Reduce the parallelism from 8 to 2 for 'flaky-tests'.
- Run `check` rather than `build` for 'flaky-tests'.
- Dump full thread dump when 'flaky-tests' is stuck.

Result:

- Closes #3742

### CODE REVIEW

**Review by minwoox — APPROVED** [review]
> It's somewhat unclear why this happens to only 'flaky-tests' job

🤔

**Comment by ikhoon** [comment]
>  Cancelled after 60m — flaky-tests 

Hmm... `flaky-tests` build did not complete in 60 minutes. 🤔

File: .github/workflows/actions_build.yml [hunk 1]
```diff
@@ -194,3 +193,8 @@ jobs:
           rm -f ~/.gradle/caches/*/*.lock || true
           rm -f ~/.gradle/caches/*/gc.properties || true
         shell: bash
+
+    - name: Dump stuck threads
+      if: always()
+      run: jps | grep -vi "jps" | awk '{ print $1 }' | xargs -I'{}' jstack -l {} || true
+      shell: bash
```
Comment (ikhoon) [inline .github/workflows/actions_build.yml:200]:
```suggestion
      - name: Dump stuck threads
        if: always()
        run: jps | grep -vi "jps" | awk '{ print $1 }' | xargs -I'{}' jstack -l {} || true
        shell: bash
```

**Comment by trustin** [comment]
Let me merge this because the current build timeout seems irrelevant and yet this fixes #3742 anyway.

**Pull request merged**